src/optimizer/query_rewrite.c

```diff
@@ -3846,10 +3865,39 @@ qo_check_primary_key_referenced_by_foreign_key_in_parent_spec (PARSER_CONTEXT *
 
       if (info.my_spec_cnt >= 1)
 	{
-	  if (curr_pred->or_next == NULL && PT_NODE_IS_EXPR (curr_pred) && PT_EXPR_OP (curr_pred) == PT_EQ
-	      && pt_is_attr (PT_EXPR_ARG1 (curr_pred)) && pt_is_attr (PT_EXPR_ARG2 (curr_pred)))
+	  if (curr_pred->or_next == NULL && PT_NODE_IS_EXPR (curr_pred) && PT_EXPR_OP (curr_pred) == PT_EQ)
 	    {
-	      reduce_pred_point_list = parser_append_node (pt_point (parser, curr_pred), reduce_pred_point_list);
+	      if (pt_is_attr (PT_EXPR_ARG1 (curr_pred)) && pt_is_attr (PT_EXPR_ARG2 (curr_pred)))
+		{
+		  reduce_pred_point_list = parser_append_node (pt_point (parser, curr_pred), reduce_pred_point_list);
+		  continue;
+		}
+
+	      /* This is to ignore the predicate of the parent added by the pt_lambda_node function.
```

Comment:
pt_lambda_node() is called in view-merge, push-predicate. I think it doesn't seem to be a problem. However, since the lambda_xxx() function is a commonly used function, I am concerned that there may be an exception case.
Instead of this, how about excluding the predicate having same column as the column of join condition as below?
```
select b.* from parent a, child b where a.fk = b.fk and a.fk =3  -- a can be removed
```

Reply:
Fixed it. Please review again.